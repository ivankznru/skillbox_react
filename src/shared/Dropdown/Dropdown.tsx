/*eslint-env es6*/
import React from 'react';
import styles from './dropdown.css';
import { useIsMounted } from '../../hooks/useIsMounted';
import ReactDOM from 'react-dom';

interface IDropdownProps {
  children: React.ReactNode;
  classNameList?: string;
}

export function Dropdown({children, classNameList}: IDropdownProps): JSX.Element {
  const node = typeof window !== 'undefined' ? document.querySelector('#dropdown_root') : undefined;
  if (!node) { return <></>; }

  return ReactDOM.createPortal((
    <div className={styles.container}>
        <div className={styles.listContainer}>
          <div className={`styles.list ${classNameList}`}>
            {children}
          </div>
        </div>
    </div>
  ), node);
}