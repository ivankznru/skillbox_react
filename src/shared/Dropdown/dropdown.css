.container {
  position: fixed;
  right: calc(10% + 50px);
  top: 20%;
  z-index: 999;
  /*
  margin: auto;
  width: 100%;
  flex: 1 1 0%;
  */
}

.listContainer {
    /*
    position: relative;
    */
}
  
.list {
  /*
  z-index: 1000;
  position: absolute;
  left: 0;
  top: 0;
  width: auto;
  width: 100%;
  height: 100%;
  */
}
