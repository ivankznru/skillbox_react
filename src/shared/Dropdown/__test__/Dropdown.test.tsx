/**
 * @jest-environment jsdom
 */
import React from 'react';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });
import { shallow } from 'enzyme';
import { Dropdown } from '../Dropdown';

describe('Dropdown', () => {
  // test('should render', () => {
  //   const wrapper = shallow(<Dropdown children={<div />} button={<button />}/>);

  //   expect(wrapper).toBeDefined();
  //   console.log(wrapper.find('div.container').debug());
  //   console.log(window);

  //   expect(wrapper.find('#button')).toBeDefined();  // wrong
  //   expect(wrapper.find('div.container').isEmptyRender()).toBeFalsy();
  // })

  // test('should render (snapshot)', () => {
  //   const wrapper = shallow(<Dropdown children={<div />} button={<button />}/>);

  //   expect(wrapper).toMatchSnapshot();
  // })
})
