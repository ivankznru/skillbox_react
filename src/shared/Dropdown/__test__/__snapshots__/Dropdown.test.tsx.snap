// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render (snapshot) 1`] = `
<div
  className="container"
>
  <div
    onBlur={[Function]}
    onClick={[Function]}
  >
    <button />
  </div>
</div>
`;
